hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -340,8 +343,7 @@ public static InetSocketAddress createSocketAddr(String target) {
   private DataNodePeerMetrics peerMetrics;
   private DataNodeDiskMetrics diskMetrics;
   private InetSocketAddress streamingAddr;
-  
-  // See the note below in incrDatanodeNetworkErrors re: concurrency.
+
   private LoadingCache<String, Map<String, Long>> datanodeNetworkCounts;
```

Comment:
I think `HashMap<String, LongAdder>` is more efficient than `ConcurrentHashMap<String, Long>` because the LongAdder instance in the Map is never replaced in this case.

Reply:
Oh I found the interface
```
  @Override // DataNodeMXBean
  public Map<String, Map<String, Long>> getDatanodeNetworkCounts() {
    return datanodeNetworkCounts.asMap();
  }
```
Since the interface cannot be changed, it's okay to use ConcurrentHashMap. +1.